Geometry/HGCalGeometry/src/HGCalGeometry.cc

```diff
@@ -312,34 +429,65 @@ unsigned int HGCalGeometry::getClosestCellIndex (const GlobalPoint& r) const {
 
   float phip = r.phi();
   float zp   = r.z();
-  unsigned int cellIndex =  m_cellVec.size();
   float dzmin(9999), dphimin(9999), dphi10(0.175);
-  for (unsigned int k=0; k<m_cellVec.size(); ++k) {
-    float dphi = phip-m_cellVec[k].phiPos();
-    while (dphi >   M_PI) dphi -= 2*M_PI;
-    while (dphi <= -M_PI) dphi += 2*M_PI;
-    if (fabs(dphi) < dphi10) {
-      float dz = fabs(zp - m_cellVec[k].getPosition().z());
-      if (dz < (dzmin+0.001)) {
-	dzmin     = dz;
-	if (fabs(dphi) < (dphimin+0.01)) {
-	  cellIndex = k;
-	  dphimin   = fabs(dphi);
-	} else {
-	  if (cellIndex >= m_cellVec.size()) cellIndex = k;
+  unsigned int cellIndex(0);
+  if (m_det == DetId::HGCalHSc) {
```

Comment:
the contents of this block can be made into a function into which `m_cellVec2` or `m_cellVec` is passed, to reduce code duplication